### ISSUE TITLE
Katib Trial is not getting succeeded even if the Metrics is captured

### ISSUE THREAD

**Issue by luckyskg** [kubeflow/katib#771]
/kind bug

**What steps did you take and what happened:**
Though katib trial execution logic is completed as evident from the Metrics being captured for the trial , the trial status is not getting succeeded. It's stuck at TrialRunning. 
Sometimes the status changes to succeeded after a while and other times status remains stuck at TrialRunning

Status:
  Conditions:
    Last Transition Time:  2019-09-18T01:13:03Z
    Last Update Time:      2019-09-18T01:13:03Z
    Message:               Trial is created
    Reason:                TrialCreated
    Status:                True
    Type:                  Created
    Last Transition Time:  2019-09-18T01:13:03Z
    Last Update Time:      2019-09-18T01:13:03Z
    Message:               Trial is running
    Reason:                TrialRunning
    Status:                True
    Type:                  Running
  Observation:
    Metrics:
      Name:    epoch_val_acc
      Value:   0.8421052694320679
  Start Time:  2019-09-18T01:13:03Z
Events:        <none>

**What did you expect to happen:**
Ideally once Metrics is captured , trial status should change to succeeded.

**Environment:**

- Kubeflow Build commit: 812ca7f
- Katib API Version: kubeflow.org/v1alpha2

**Comment by johnugeorge** [comment]
Can you paste your config? What is the status of underlying job?

**Comment by johnugeorge** [comment]
What are the steps to reproduce the issue

**Comment by luckyskg** [comment]
Name:         cnn-exp-d002e0d5-5swrc4bv
Namespace:    kubeflow
Labels:       experiment=cnn-exp-d002e0d5
Annotations:  <none>
API Version:  kubeflow.org/v1alpha2
Kind:         Trial
Metadata:
  Creation Timestamp:  2019-09-18T01:13:03Z
  Generation:          1
  Owner References:
    API Version:           kubeflow.org/v1alpha2
    Block Owner Deletion:  true
    Controller:            true
    Kind:                  Experiment
    Name:                  cnn-exp-d002e0d5
    UID:                   76800110-d9b1-11e9-8d71-42010a8001a0
  Resource Version:        23021511
  Self Link:               /apis/kubeflow.org/v1alpha2/namespaces/kubeflow/trials/cnn-exp-d002e0d5-5swrc4bv
  UID:                     769a233d-d9b1-11e9-8d71-42010a8001a0
Spec:
  Metrics Collector Spec:  apiVersion: batch/v1beta1
kind: CronJob
metadata:
  name: cnn-exp-d002e0d5-5swrc4bv
  namespace: kubeflow
spec:
  schedule: "*/1 * * * *"
  successfulJobsHistoryLimit: 0
  failedJobsHistoryLimit: 1
  jobTemplate:
    spec:
      template:
        spec:
          containers:
          - name: cnn-exp-d002e0d5-5swrc4bv
            image: codebreach/tfevent-metricscollector:latest
            args:
            - "python"
            - "main.py"
            - "-t"
            - "cnn-exp-d002e0d5-5swrc4bv"
            - "-d"
            - "gs://dummy/cnn_test/mu_q09/hyperparam_tuning/cnn-exp-d002e0d5-5swrc4bv/log_dir"
            - "-m"
            - "epoch_acc,epoch_val_acc"
            volumeMounts:
              - name: google-cloud-key
                mountPath: /var/secrets/google
            env:
            - name: GOOGLE_APPLICATION_CREDENTIALS
              value: /var/secrets/google/key.json
          volumes:
          - name: google-cloud-key
            secret:
              secretName: google-cloud-key
          restartPolicy: Never
          serviceAccountName: metrics-collector
  Objective:
    Goal:                   0.99
    Objective Metric Name:  epoch_val_acc
    Type:                   maximize
  Parameter Assignments:
    Name:    parameter_num_filters
    Value:   64
    Name:    parameter_filter_sizes
    Value:   3,4,5
    Name:    embedding_dim
    Value:   100
  Run Spec:  apiVersion: batch/v1
kind: Job
metadata:
  name: cnn-exp-d002e0d5-5swrc4bv
  namespace: kubeflow
spec:
  template:
    spec:
      volumes:
      - name: google-cloud-key
        secret:
          secretName: google-cloud-key
      imagePullSecrets:
      - name: gcr-io-secret
      containers:
      - name: cnn-exp-d002e0d5-5swrc4bv
        image: gcr.io/dummy/cnn_multi_label_multi_class_train:cnn-multi-label_20190916_114558_ff5b44c4
        args:
        - "--train_file=gs://dummy/cnn_test/mu_q09/train.csv"
        - "--glove_dir=gs://dummy/glove_embeddings"
        - "--parameter_num_epochs=100"
        - "--parameter_batch_size=32"
        - "--output_dir=gs://dummy/mu_q09/hyperparam_tuning/cnn-exp-d002e0d5-5swrc4bv"
        - "--text_column_name=replaced_text"
        - "--label_column_name=selected_value"
        - "--es_monitoring_metric=val_loss"
        - "--es_mode=auto"
        - "--es_min_delta=0.0005"
        - "--es_patience=5"
        - "--type_of_classification=multi_label"
        - "--is_trainable=1"
        - "--parameter_spatial_dropout=0.2"
        - "--parameter_num_filters=64"
        - "--parameter_filter_sizes=3,4,5"
        - "--embedding_dim=100"
        env:
        - name: GOOGLE_APPLICATION_CREDENTIALS
          value: /var/secrets/google/key.json
        - name: INTRA_OP_PARALLELISM_THREADS
          value: "6"
        - name: INTER_OP_PARALLELISM_THREADS
          value: "6"
        volumeMounts:
        - name: google-cloud-key
          mountPath: /var/secrets/google
        resources:
          limits:
            cpu: 4
            memory: 9Gi
          memory:
            cpu: 4
            memory: 9Gi
      restartPolicy: Never
Status:
  Conditions:
    Last Transition Time:  2019-09-18T01:13:03Z
    Last Update Time:      2019-09-18T01:13:03Z
    Message:               Trial is created
    Reason:                TrialCreated
    Status:                True
    Type:                  Created
    Last Transition Time:  2019-09-18T01:13:03Z
    Last Update Time:      2019-09-18T01:13:03Z
    Message:               Trial is running
    Reason:                TrialRunning
    Status:                True
    Type:                  Running
  Observation:
    Metrics:
      Name:    epoch_val_acc
      Value:   0.8421052694320679
  Start Time:  2019-09-18T01:13:03Z
Events:        <none>

**Comment by luckyskg** [comment]
I had added step in kubeflow pipeline which trains a CNN model using Katib.
First Issue is It takes around 20 minutes to schedule pod for the katib experiment in kubenetes cluster
Second Issue is even after Trial has picked the metrics from the run, the status never changes to succeeded.


The following is the job config

apiVersion: v1
kind: Pod
metadata:
  annotations:
    workflows.argoproj.io/node-name: cnn-pipeline-4ss57.exit-handler-1.cnn-evaluation
    workflows.argoproj.io/template: '{"name":"cnn-evaluation","inputs":{"parameters":[{"name":"cnn-katib-experiment-best-trial","value":"cnn-exp-16c7ce8c-qp5js42x"},{"name":"cpu-limit","value":"3"},{"name":"eval-size","value":"-1"},{"name":"input-file","value":"gs://dummy/selected_value_prediction/cnn_test/mu_q09/train.csv"},{"name":"label-column-name","value":"selected_value"},{"name":"memory-limit","value":"9Gi"},{"name":"output","value":"gs://dummy/selected_value_prediction/cnn_test/mu_q09"},{"name":"question-text","value":"mu_q09"},{"name":"text-column-name","value":"replaced_text"}]},"outputs":{"parameters":[{"name":"cnn-evaluation-ensemble-model-path","valueFrom":{"path":"/ensemble_model_path.txt"}},{"name":"cnn-evaluation-results","valueFrom":{"path":"/predictions.txt"}}],"artifacts":[{"name":"mlpipeline-ui-metadata","path":"/mlpipeline-ui-metadata.json","optional":true},{"name":"mlpipeline-metrics","path":"/mlpipeline-metrics.json","optional":true}]},"metadata":{},"container":{"name":"","image":"gcr.io/dummy/text_models/cnn_eval:kube-cnn_20190917_055601_4b6734b7","args":["--input_file","gs://dummy/selected_value_prediction/cnn_test/mu_q09/train.csv","--feature_cols","replaced_text","--label_col","selected_value","--question","mu_q09","--model_dir","gs://dummy/selected_value_prediction/cnn_test/mu_q09/predict/cnn-exp-16c7ce8c-qp5js42x","--output","gs://dummy/selected_value_prediction/cnn_test/mu_q09/e9d40b22-d9bc-11e9-8d71-42010a8001a0/data","--eval_size","-1"],"env":[{"name":"GOOGLE_APPLICATION_CREDENTIALS","value":"/secret/gcp-credentials/key.json"},{"name":"CLOUDSDK_AUTH_CREDENTIAL_FILE_OVERRIDE","value":"/secret/gcp-credentials/key.json"},{"name":"RESOURCE_CPU_LIMIT","value":"3"},{"name":"RESOURCE_MEMORY_LIMIT","value":"9Gi"}],"resources":{},"volumeMounts":[{"name":"gcp-credentials-gcs-credentials","mountPath":"/secret/gcp-credentials"}]},"sidecars":[{"name":"predict-api","image":"gcr.io/dummy/text_models/cnn_w2v__predict:cnn-multi-label_20190916_131428_dc2991ce","args":["--model_dir","gs://dummy/selected_value_prediction/cnn_test/mu_q09/predict/cnn-exp-16c7ce8c-qp5js42x","--model_type","cnn_ml_mc"],"resources":{"limits":{"cpu":"3","memory":"12Gi"}}}],"archiveLocation":{"s3":{"endpoint":"minio-service.kubeflow:9000","bucket":"mlpipeline","insecure":true,"accessKeySecret":{"name":"mlpipeline-minio-artifact","key":"accesskey"},"secretKeySecret":{"name":"mlpipeline-minio-artifact","key":"secretkey"},"key":"artifacts/cnn-pipeline-4ss57/cnn-pipeline-4ss57-2927179935"}}}'
  creationTimestamp: 2019-09-18T03:06:50Z
  labels:
    workflows.argoproj.io/completed: "false"
    workflows.argoproj.io/workflow: cnn-pipeline-4ss57
  name: cnn-pipeline-4ss57-2927179935
  namespace: kubeflow
  ownerReferences:
  - apiVersion: argoproj.io/v1alpha1
    blockOwnerDeletion: true
    controller: true
    kind: Workflow
    name: cnn-pipeline-4ss57
    uid: e9d40b22-d9bc-11e9-8d71-42010a8001a0
  resourceVersion: "23051418"
  selfLink: /api/v1/namespaces/kubeflow/pods/cnn-pipeline-4ss57-2927179935
  uid: 5c12f479-d9c1-11e9-8d71-42010a8001a0
spec:
  containers:
  - command:
    - argoexec
    - wait
    env:
    - name: ARGO_POD_NAME
      valueFrom:
        fieldRef:
          apiVersion: v1
          fieldPath: metadata.name
    image: argoproj/argoexec:v2.3.0
    imagePullPolicy: IfNotPresent
    name: wait
    resources: {}
    terminationMessagePath: /dev/termination-log
    terminationMessagePolicy: File
    volumeMounts:
    - mountPath: /argo/podmetadata
      name: podmetadata
    - mountPath: /var/run/docker.sock
      name: docker-sock
      readOnly: true
    - mountPath: /argo/secret/mlpipeline-minio-artifact
      name: mlpipeline-minio-artifact
      readOnly: true
    - mountPath: /mainctrfs/secret/gcp-credentials
      name: gcp-credentials-gcs-credentials
    - mountPath: /var/run/secrets/kubernetes.io/serviceaccount
      name: pipeline-runner-token-dpjtk
      readOnly: true
  - args:
    - --input_file
    - gs://dummy/selected_value_prediction/cnn_test/mu_q09/train.csv
    - --feature_cols
    - replaced_text
    - --label_col
    - selected_value
    - --question
    - mu_q09
    - --model_dir
    - gs://dummy/selected_value_prediction/cnn_test/mu_q09/predict/cnn-exp-16c7ce8c-qp5js42x
    - --output
    - gs://dummy/selected_value_prediction/cnn_test/mu_q09/e9d40b22-d9bc-11e9-8d71-42010a8001a0/data
    - --eval_size
    - "-1"
    env:
    - name: GOOGLE_APPLICATION_CREDENTIALS
      value: /secret/gcp-credentials/key.json
    - name: CLOUDSDK_AUTH_CREDENTIAL_FILE_OVERRIDE
      value: /secret/gcp-credentials/key.json
    - name: RESOURCE_CPU_LIMIT
      value: "3"
    - name: RESOURCE_MEMORY_LIMIT
      value: 9Gi
    image: gcr.io/dummy/text_models/cnn_eval:kube-cnn_20190917_055601_4b6734b7
    imagePullPolicy: IfNotPresent
    name: main
    resources: {}
    terminationMessagePath: /dev/termination-log
    terminationMessagePolicy: File
    volumeMounts:
    - mountPath: /secret/gcp-credentials
      name: gcp-credentials-gcs-credentials
    - mountPath: /var/run/secrets/kubernetes.io/serviceaccount
      name: pipeline-runner-token-dpjtk
      readOnly: true
  - args:
    - --model_dir
    - gs://dummy/selected_value_prediction/cnn_test/mu_q09/predict/cnn-exp-16c7ce8c-qp5js42x
    - --model_type
    - cnn_ml_mc
    image: gcr.io/dummy/text_models/cnn_w2v__predict:cnn-multi-label_20190916_131428_dc2991ce
    imagePullPolicy: IfNotPresent
    name: predict-api
    resources:
      limits:
        cpu: "3"
        memory: 12Gi
      requests:
        cpu: "3"
        memory: 12Gi
    terminationMessagePath: /dev/termination-log
    terminationMessagePolicy: File
    volumeMounts:
    - mountPath: /var/run/secrets/kubernetes.io/serviceaccount
      name: pipeline-runner-token-dpjtk
      readOnly: true
  dnsPolicy: ClusterFirst
  imagePullSecrets:
  - name: gcr_io_secret
  nodeName: gke-kubeflow-preemptible-4-59d2d393-9r79
  priority: 0
  restartPolicy: Never
  schedulerName: default-scheduler
  securityContext: {}
  serviceAccount: pipeline-runner
  serviceAccountName: pipeline-runner
  terminationGracePeriodSeconds: 30
  tolerations:
  - effect: NoExecute
    key: node.kubernetes.io/not-ready
    operator: Exists
    tolerationSeconds: 300
  - effect: NoExecute
    key: node.kubernetes.io/unreachable
    operator: Exists
    tolerationSeconds: 300
  volumes:
  - downwardAPI:
      defaultMode: 420
      items:
      - fieldRef:
          apiVersion: v1
          fieldPath: metadata.annotations
        path: annotations
    name: podmetadata
  - hostPath:
      path: /var/run/docker.sock
      type: Socket
    name: docker-sock
  - name: gcp-credentials-gcs-credentials
    secret:
      defaultMode: 420
      secretName: gcs-credentials
  - name: mlpipeline-minio-artifact
    secret:
      defaultMode: 420
      items:
      - key: accesskey
        path: accesskey
      - key: secretkey
        path: secretkey
      secretName: mlpipeline-minio-artifact
  - name: pipeline-runner-token-dpjtk
    secret:
      defaultMode: 420
      secretName: pipeline-runner-token-dpjtk
status:
  conditions:
  - lastProbeTime: null
    lastTransitionTime: 2019-09-18T03:06:50Z
    status: "True"
    type: Initialized
  - lastProbeTime: null
    lastTransitionTime: 2019-09-18T03:07:56Z
    message: 'containers with unready status: [predict-api]'
    reason: ContainersNotReady
    status: "False"
    type: Ready
  - lastProbeTime: null
    lastTransitionTime: 2019-09-18T03:07:56Z
    message: 'containers with unready status: [predict-api]'
    reason: ContainersNotReady
    status: "False"
    type: ContainersReady
  - lastProbeTime: null
    lastTransitionTime: 2019-09-18T03:06:50Z
    status: "True"
    type: PodScheduled
  containerStatuses:
  - containerID: docker://b0020750cffad70adc4c03af20b3bb506004ce262ebe7773d302ddded58e463f
    image: gcr.io/dummy/text_models/cnn_eval:kube-cnn_20190917_055601_4b6734b7
    imageID: docker-pullable://gcr.io/dummy/text_models/cnn_eval@sha256:9075a8e405f1553e2671937d21b2643aa56f5f40502d28de5d2aa61f90bc9e91
    lastState: {}
    name: main
    ready: true
    restartCount: 0
    state:
      running:
        startedAt: 2019-09-18T03:07:09Z
  - containerID: docker://d2174cbd5a612b4021b00cca9c412e75a7283558964c8f6eaafeb2de4d1d8119
    image: gcr.io/dummy/text_models/cnn_w2v__predict:cnn-multi-label_20190916_131428_dc2991ce
    imageID: docker-pullable://gcr.io/dummy/text_models/cnn_w2v__predict@sha256:af8b3728195369360fbb21526abbfd055fe9a05fa3e088f83ec832011b945f99
    lastState: {}
    name: predict-api
    ready: false
    restartCount: 0
    state:
      terminated:
        containerID: docker://d2174cbd5a612b4021b00cca9c412e75a7283558964c8f6eaafeb2de4d1d8119
        exitCode: 1
        finishedAt: 2019-09-18T03:07:55Z
        reason: Error
        startedAt: 2019-09-18T03:07:41Z
  - containerID: docker://8a9c0046a36289069574cbb0e511380694b5a0604d05c2cd8a53c57e04e05844
    image: argoproj/argoexec:v2.3.0
    imageID: docker-pullable://argoproj/argoexec@sha256:85132fc2c8bc373fca885df17637d5d35682a23de8d1390668a5e1c149f2f187
    lastState: {}
    name: wait
    ready: true
    restartCount: 0
    state:
      running:
        startedAt: 2019-09-18T03:06:51Z
  hostIP: 10.128.0.6
  phase: Running
  podIP: 10.11.63.5
  qosClass: Burstable
  startTime: 2019-09-18T03:06:50Z

**Comment by johnugeorge** [comment]
Can you check the status of Jobs ? I suspect that jobs haven’t been terminated

**Comment by johnugeorge** [comment]
Any update on this?

**Comment by luckyskg** [comment]
Unfortunately the job was deleted. And i am not able to reproduce the issue with sub sequent runs. So we can close the issue
